lib/src/rules/require_trailing_commas.dart

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/token.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import '../analyzer.dart';
+
+const _desc = r'Use trailing commas for all function calls and declarations.';
+
+const _details = r'''
+
+**DO** use trailing commas for all function calls and declarations unless the
+function call or definition, from the start of the function name up to the
+closing parenthesis, fits in a single line.
+
+**GOOD:**
+```dart
+void run() {
+  method(
+    'does not fit on one line',
+    'test test test test test test test test test test test',
+  );
+}
+```
+
+**BAD:**
+```dart
+void run() {
+  method('does not fit on one line',
+      'test test test test test test test test test test test');
+}
+```
+
+**Exception:** If the final parameter/argument is positional (vs named) and is
+either a function literal implemented using curly braces, a literal map, a
+literal set or a literal array. This exception only applies if the final
+parameter does not fit entirely on one line.
+
+**Note:** This lint rule assumes `dartfmt` has been run over the code.
```

Comment:
Consider adding "and might produce false positives until that's happened" to make the implications more clear.

Reply:
Updated, although I feel that this could be a given.

The assumption that `dartfmt` has been run is not unique to this lint rule.

[`curly_braces_in_flow_control_structures`](https://github.com/dart-lang/linter/blob/master/lib/src/rules/curly_braces_in_flow_control_structures.dart) also has similar assumptions but is undocumented.